File: angularclient/src/app/model/Robots/RobotBattery.ts
import {BatteryType} from "../type/BatteryType";
export class RobotBattery{
  id: string;
  name: string;
  type: BatteryType;

  constructor(name: string = '', type: BatteryType = new BatteryType()) {
    this.name = name;
    this.type = type;
  }


}

File: ra-back/src/main/java/pl/raptors/ra_back/domain/robots/RobotBattery.java
package pl.raptors.ra_back.domain.robots;

import lombok.Data;

import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@NoArgsConstructor
@Data
@Document(collection = "batteries")
public class RobotBattery {

    @Id
    private String id;
    //private Date productionDate;
    private String productionDate; //w stringu będzie to zapisywane ze wględu na możliwość czytelniejszego przechoywywania i formatowania daty
    private BatteryType type;

    public RobotBattery(String productionDate, BatteryType type)
    {
        this.productionDate = productionDate;
        this.type = type;
    }
}
